### PR TITLE
Enhance vnet_vxlan for scaling of vnets and routes

### PR DESCRIPTION
**Pull request by manjuv1**
### Description of PR

Summary:
Enhance vnet_vxlan for scaling of Vnets/Routes/Endpoints

### Type of change

- [] Bug fix
- [] Testbed and Framework(new/improvement)
- [Y] Test case(new/improvement)

#### How did you do it?
Enhance vnet_vxlan test to scale to a configured number of vnets/routes/endpoints. If none of these configurations are passed, the default 8 Vnets will be created.

#### How did you verify/test it?
ansible-playbook (generated vnet_config.yml)
Pass the new parameters to ansible-playbook for the scale test: -e "num_vnet=51 num_routes=550 num_endpoints=200"
For the existing test with 8 Vnets, no need to pass the above parameters.

### CODE REVIEW

File: ansible/roles/test/templates/vnet_config.j2 [hunk 1]
```diff
@@ -25,100 +98,162 @@ vnet_id_list:
...
-  - Vnet3: Vnet4
-  - Vnet4: Vnet3
-  - Vnet7: Vnet8
-  - Vnet8: Vnet7
+{% for vnet_batch in range (1, 1 + num_vnet)|batch(4) %}
+  - Vnet{{ vnet_batch[2] }}: Vnet{{ vnet_batch[3] }}
+  - Vnet{{ vnet_batch[3] }}: Vnet{{ vnet_batch[2] }}
+{% endfor %}
 
+{# Vnet - Interface Configurations #}
```
Comment (prsunny) [inline ansible/roles/test/templates/vnet_config.j2]: Can we have this set to max `128`?

Comment (manjuv1) [reply ansible/roles/test/templates/vnet_config.j2]: Can you please review if the changes to limit to 128 are fine? I still have to validate and see if the total routes exceeds the configured "num_routes" with the introduction of new sets of peers.

Comment (prsunny) [reply ansible/roles/test/templates/vnet_config.j2]: Changes looks ok to me.

**Comment by volodymyrsamotiy** [comment]
After latest changes ```Serv2Serv``` works good now, but found another issue that in some cases not all required neighbors are configured correctly and as a result test fails.
Issue is observed when try to check 128 VNETs, when run playbook with the ```num_vnet=128```, for example:
```
$ ansible-playbook test_sonic.yml -i inventory -e testbed_name=<testbed_name> -e testbed_type=t0 -e testcase_name=vnet_vxlan -e "num_vnet=128 num_routes=1024 num_endpoints=512"
```
After test failed go to DUT and check the following:
```
root@sonic:/home/admin# cat /tmp/vnet.nbr.json | grep Vlan | wc -l
192
root@sonic:/home/admin# redis-cli -n 4 keys "NEIGH*Vlan*" | wc -l
192
root@sonic:/home/admin# redis-cli -n 0 keys "NEIGH*Vlan*" | wc -l
37
```
As shown above test generates config ```vnet.nbr.json``` with the 192 neighbors to be configured and we do observe 192 entries in ```CONFIG_DB``` but to ```APPLICATION_DB``` only 37 neighbors was configured.
As a result some test cases are failing since many neighbors are missed and there is no way for traffic to pass.

**Pull request merged**